parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -851,6 +860,8 @@ pub mod pallet {
 				message.proof.block_hash,
 			);
 
+			<LatestVerifiedMessageBlock<T>>::set(block_number);
```

Comment:
This approach is gonna be problematic, as it ignores that the basic and incentivized inbound channel pallets have their own logical message stream.

Depending on the message, the relayer will submit it either to the basic or incentivized inbound channel pallet.

So the problem that occurs in your approach is the following: Suppose the basic inbound pallet last processed a message at block 100.

A relayer submits incentivized messages to the incentivized channel, all the way up to block 10000. It ignores basic messages. Now `LatestVerifiedMessageBlock` points to block 1000, even though the basic inbound channel still needs to be backfed messages from 100-1000.

So actually we need to track LatestVerifiedMessageBlock separately for the basic and incentivized channels.

Reply:
> So actually we need to track LatestVerifiedMessageBlock separately for the basic and incentivized channels.

And likely the relayer would have 2 separate goroutines that backfill messages to the basic channel or the incentivized channel.

Reply:
I reverted the message backfill, will open another PR then.

Reply:
@vgeddes so I've been thinking about this one... Both channel events are queried using the same start and end block range. So if the events are ordered by block number and then processed, I think the scenario you described above won't happen.

This does add requirements to the relayer (to sort the events by block number), which goes against the principle of keeping the relayer as dumb as possible. So I could still add tracking the last verified message separately per channel. Do you envision having more than 2 channels for the bridge in the future?

Reply:
One thing that I am also thinking about is that I don't know from which channel the message came in the beacon light client. Could I add that as a parameter to the `verify` method? I could perhaps send the whole message ID (channel and nonce)?